pytorch_lightning/trainer/trainer.py

```diff
@@ -490,13 +490,48 @@ def _setup_on_init(self, num_sanity_val_steps: int) -> None:
 
         self.num_predict_batches = []
 
+    def _call_and_handle_interrupt(self, trainer_fn, *args, **kwargs):
```

Comment:
Add typing

Reply:
please also add a comment/docstring that this is intended to be used only for main trainer function entry points (fit, validate, test, predict) as all errors should funnel through this